test/integration/smoke/test_activate_template.py

```diff
@@ -0,0 +1,196 @@
+    def setUpClass(cls):
+        cls.testClient = super(TestActivateTemplate, cls).getClsTestClient()
+        cls.apiclient = cls.testClient.getApiClient()
+        cls.dbclient = cls.testClient.getDbConnection()
+        cls._cleanup = []
+        cls.templates = []
+
+        cls.services = cls.testClient.getParsedTestDataConfig()
+        cls.unsupportedHypervisor = False
+        cls.hypervisor = cls.testClient.getHypervisorInfo()
+        if cls.hypervisor.lower() not in ['kvm']:
+            # Only testing kvm
+            cls.unsupportedHypervisor = True
+            return
+
+        # Get Zone, Domain and templates
+        cls.domain = get_domain(cls.apiclient)
+        cls.zone = get_zone(cls.apiclient, cls.testClient.getZoneForTests())
+        cls.services["mode"] = cls.zone.networktype
+        cls.services["virtual_machine"]["zoneid"] = cls.zone.id
+        cls.account = Account.create(
+            cls.apiclient,
+            cls.services["account"],
+            admin=True,
+            domainid=cls.domain.id
+        )
+        cls._cleanup.append(cls.account)
+        cls.user = Account.create(
+            cls.apiclient,
+            cls.services["account"],
+            domainid=cls.domain.id
+        )
+        cls._cleanup.append(cls.user)
+        cls.service_offering = ServiceOffering.create(
+            cls.apiclient,
+            cls.services["service_offerings"]["tiny"]
+        )
+        cls._cleanup.append(cls.service_offering)
+    
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    def setUp(self):
+        # Official cloudstack system vm template
+        self.test_template = registerTemplate.registerTemplateCmd()
+        self.test_template.hypervisor = self.hypervisor
+        self.test_template.zoneid = self.zone.id
+        self.test_template.name = 'test-system-kvm-4.11.3'
+        self.test_template.displaytext = 'test-system-kvm-4.11.3'
+        self.test_template.url = "http://download.cloudstack.org/systemvm/4.11/systemvmtemplate-4.11.3-kvm.qcow2.bz2"
+        self.test_template.format = "QCOW2"
+        self.test_template.system = True
+        self.test_template.ostypeid = self.getOsType("Debian GNU/Linux 9 (64-bit)")
+        self.md5 = "d40bce40b2d5bb4ba73e56d1e95aeae5"
+
+        self.activateTemplateCmd = activateSystemVMTemplate.activateSystemVMTemplateCmd()
+        self.listConfigurationsCmd = listConfigurations.listConfigurationsCmd()
+        self.listConfigurationsCmd.name = 'router.template.kvm'
+
+        self.getDefaultURLCmd = getSystemVMTemplateDefaultUrl.getSystemVMTemplateDefaultUrlCmd()
+        self.getDefaultURLCmd.hypervisor = 'KVM'
+
+        return
+
+    @attr(tags=["advanced", "smoke"], required_hardware="true")
+    def test_01_activate_sytem_vm_template(self):
```

Comment:
This test needs to be revering the old value properly, since it'll brake any further test execution. OK it registers a template and sets it as default template, I can live with this simple check (not going into different hypervisors and versions), but.. if any other test tries to deploy a router with cloudstack requiring different version of template it'll fail simply because it's using hardcoded 4.11.3 one. 
We have the option to make this dynamic ie call the api to get default template url
`(localcloud) SBCM5> > get systemvmtemplatedefaulturl hypervisor=kvm
{
  "url": {
    "url": "https://download.cloudstack.org/systemvm/4.11/systemvmtemplate-4.11.3-kvm.qcow2.bz2"
  }
}`
and go with it

Reply:
Done